src/drafts/SelectPanel2/SelectPanel.tsx

```diff
@@ -256,23 +273,26 @@ const SelectPanelFooter = ({...props}) => {
         borderColor: 'border.default',
       }}
     >
-      <div>{props.children}</div>
-      <Box sx={{display: 'flex', gap: 2}}>
-        <Button size="small" type="button" onClick={() => onCancel()}>
-          Cancel
-        </Button>
-        <Button size="small" type="submit" variant="primary">
-          Save
-        </Button>
-      </Box>
+      <Box sx={{flexGrow: hidePrimaryActions ? 1 : 0}}>{props.children}</Box>
+
+      {hidePrimaryActions ? null : (
+        <Box sx={{display: 'flex', gap: 2}}>
+          <Button size="small" type="button" onClick={() => onCancel()}>
+            Cancel
+          </Button>
+          <Button size="small" type="submit" variant="primary">
+            Save
+          </Button>
+        </Box>
+      )}
     </Box>
   )
 }
 SelectPanel.Footer = SelectPanelFooter
 
 // @ts-ignore todo
 SelectPanel.SecondaryButton = props => {
-  return <Button {...props} size="small" type="button" />
+  return <Button {...props} size="small" type="button" block />
```

Comment:
Just wondering why this `Button` needs to be block type?

Reply:
Good question! 

So, for most cases of SelectPanel you'd see a Save and Cancel button on the right, but then also an optional secondary button on the left:

<img width="300" alt="select panel with save, cancel and a secondary action in the footer" src="https://github.com/primer/react/assets/1863771/e2a4393a-1e72-4cdb-8f24-21a5c2c1c931">

But, in the case of instant variant, there is no Save and Cancel, so we'd like the secondary button to take the full width of the footer:

Without block:

<img width="346" alt="select panel with just a secondary action in the footer, awkwardly sitting in the left corner" src="https://github.com/primer/react/assets/1863771/2c02e468-20b9-4590-a3f4-3b49770347b0">

With block:

<img width="350" alt="select panel with just a secondary action in the footer, taking the full width of the footer" src="https://github.com/primer/react/assets/1863771/245ba6ee-480e-473a-b342-f523d888393d">

Now because the footer has 2 divs inside of it - left container and right container, it's always safe for the secondary button to take full width (`block`) of the left container.

![image](https://github.com/primer/react/assets/1863771/2669a79a-7769-45c1-b522-c94230144ba4)

Reply:
I see! thanks for the explainer!